### ISSUE TITLE
A11y / Sensei / Learn - Quiz questions not associated with answer options

### ISSUE THREAD

**Issue by joedolson** [WordPress/Learn#2779]
## Type of feedback
//dev

## Description

The input type for multiple choice questions are a radio group, but there is no semantic grouping to associate the question with the answers. For a screen reader user navigating using tab, this means that extra exploration is required to find out what question you're answering. 

The structure should be either a `fieldset` where the question is the `legend` value, or a group with role `radiogroup` and an `aria-labelledby` attribute pointing to the ID of the question.

## Screenshots or screen recording (optional)

![image](https://github.com/user-attachments/assets/aceba4cb-3a00-4081-8fa3-5d3b7bb4e880)
